crates/sequencer/src/sequencer.rs

```diff
@@ -541,6 +558,15 @@ where
 
                 self.mempool.update_accounts(account_updates);
 
+                // If we exceed the threshold, we should notify the commitment
+                // worker to initiate a commitment.
+                #[allow(clippy::collapsible_if)]
+                if serialized_state_diff.len() as u64 > STATEDIFF_SIZE_COMMITMENT_THRESHOLD {
```

Comment:
I think the logic should be:

```
if last_state_diff + new_state_diff > 400k {
   commit last_state_diff
   current_diff = new_state_diff
} else {
   current_diff = last_state_diff + new_state_diff
}
```

Reply:
That's done somewhere else.

Reply:
where?

Reply:
https://github.com/chainwayxyz/citrea/pull/854/files#diff-2039c1f53f29d88ff08ce7de70335902799a60dec1fcf87ba0f54ec87a292745R490

Reply:
Doesn't Roman's control flow make more sense?

Reply:
No because:
```
commit last_state_diff
current_diff = new_state_diff
```

The commit part happens asynchronously. Which means that we override `current_diff` in this version after we send the notification over the commitment channel. When the commitment channel receives the message, we will have overridden current diff with the very small `new_state_diff`

Reply:
I understand it goes async, however we may end up with > 400kb because of how it is implemented. No?

Reply:
Changed logic to how Roman suggested. I was misunderstanding some details about how this suggestion would work.